src/wui/load_or_save_game.cc

```diff
@@ -361,213 +333,180 @@ UI::Button* LoadOrSaveGame::delete_button() {
+			   (boost::format(_("Today, %1%:%2%")) % savedate->tm_hour % minute).str();
+			gamedata.savedonstring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    hour:minute. This is part of a list. */
+			   (boost::format(_("saved today at %1%:%2%")) % savedate->tm_hour % minute).str();
+		} else if ((savedate->tm_year == current_year && savedate->tm_mon == current_month &&
+		            savedate->tm_mday == current_day - 1) ||
+		           (savedate->tm_year == current_year - 1 && savedate->tm_mon == 11 &&
+		            current_month == 0 && savedate->tm_mday == 31 &&
+		            current_day == 1)) {  // Yesterday
+			// Adding the 0 padding in a separate statement so translators won't have to deal
+			// with it
+			const std::string minute = (boost::format("%02u") % savedate->tm_min).str();
+
+			gamedata.savedatestring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    hour:minute */
+			   (boost::format(_("Yesterday, %1%:%2%")) % savedate->tm_hour % minute).str();
+			gamedata.savedonstring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    hour:minute. This is part of a list. */
+			   (boost::format(_("saved yesterday at %1%:%2%")) % savedate->tm_hour % minute).str();
+		} else {  // Older
+			gamedata.savedatestring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    month day, year */
+			   (boost::format(_("%1% %2%, %3%")) % localize_month(savedate->tm_mon) %
+			    savedate->tm_mday % (1900 + savedate->tm_year))
+			      .str();
+			gamedata.savedonstring =
+			   /** TRANSLATORS: Display date for choosing a savegame/replay. Placeholders are:
+			                                                    month (short name) day (number), year
+			      (number). This is part of a list. */
+			   (boost::format(_("saved on %1% %2%, %3%")) % savedate->tm_mday %
+			    localize_month(savedate->tm_mon) % (1900 + savedate->tm_year))
+			      .str();
+		}
+	}
+}
 
-	FilenameSet gamefiles;
-	if (filetype_ == FileType::kReplay) {
-		gamefiles = g_fs->filter_directory(kReplayDir, [](const std::string& fn) {
-			return boost::algorithm::ends_with(fn, kReplayExtension);
-		});
-		// Update description column title for replays
-		table_.set_column_tooltip(2, show_filenames_ ? _("Filename: Map name (start of replay)") :
-		                                               _("Map name (start of replay)"));
-	} else {
-		gamefiles = g_fs->filter_directory(kSaveDir, [](const std::string& fn) {
-			return boost::algorithm::ends_with(fn, kSavegameExtension);
-		});
+void LoadOrSaveGame::add_general_information(SavegameData& gamedata,
```

Comment:
This function is called only once and add_time_info as well, directly after each other. Unify into 1 function?

Reply:
Sry, I disagree. Functions should do one thing only which then can be properly named (like add_time_info). Admittedly, "add_general_information is a rather poor name but still better to keep them seperated...

Reply:
Let's keep it like it is then